04.ls/ls_command.rb

```diff
@@ -1,11 +1,24 @@
 #!/usr/bin/env ruby
 # frozen_string_literal: true
 
+require 'optparse'
+
 COL_COUNT = 3
 
 def main
+  opt = OptionParser.new
+
+  options = {}
+  opt.on('-a') { |v| options[:a] = v }
+  opt.parse(ARGV)
+
   all_files = find_all_files
-  visible_files = select_visible_files(all_files)
+  visible_files =
+    if !options[:a]
+      select_visible_files(all_files)
+    else
+      all_files
+    end
```

Comment:
```suggestion
  visible_files = options[:a] ? all_files : select_visible_files(all_files)
```

条件分岐は否定よりも肯定の方が読みやすいです。（if/elseの構造になる場合はなおさら）

あと、ここは三項演算子を使った方が行数も減って良いと思います。